examples/3d/split_screen.rs

```diff
@@ -50,81 +50,65 @@ fn setup(
-                    width: Val::Percent(100.),
-                    height: Val::Percent(100.),
-                    ..default()
-                },
-                ..default()
-            },
-        ))
-        .with_children(|parent| {
-            parent.spawn(TextBundle::from_section(
-                "Left",
-                TextStyle {
-                    font_size: 20.,
+    // Cameras and their dedicated UI
+    for (index, (camera_name, camera_pos)) in [
+        ("Player 1", Vec3::new(0.0, 200.0, -150.0)),
+        ("Player 2", Vec3::new(150.0, 150., 50.0)),
+        ("Player 3", Vec3::new(100.0, 150., -150.0)),
+        ("Player 4", Vec3::new(-100.0, 80., 150.0)),
+    ]
+    .iter()
+    .enumerate()
+    {
+        let camera = commands
+            .spawn((
+                Camera3dBundle {
+                    transform: Transform::from_translation(*camera_pos)
+                        .looking_at(Vec3::ZERO, Vec3::Y),
+                    camera: Camera {
+                        // Renders cameras with different priorities to prevent ambiguities
+                        order: index as isize,
+                        // Don't clear after the first camera because the first camera already cleared the entire window
+                        clear_color: if index > 0 {
+                            ClearColorConfig::None
+                        } else {
+                            ClearColorConfig::default()
+                        },
+                        ..default()
+                    },
                     ..default()
                 },
-            ));
-            buttons_panel(parent);
-        });
-
-    commands
-        .spawn((
-            TargetCamera(right_camera),
-            NodeBundle {
-                style: Style {
-                    width: Val::Percent(100.),
-                    height: Val::Percent(100.),
-                    ..default()
+                CameraPosition {
+                    pos: UVec2::new((index % 2) as u32, (index / 2) as u32),
                 },
-                ..default()
-            },
-        ))
-        .with_children(|parent| {
-            parent.spawn(TextBundle::from_section(
-                "Right",
-                TextStyle {
-                    font_size: 20.,
+            ))
+            .id();
+
+        // Set up UI
+        commands
+            .spawn((
+                TargetCamera(camera),
+                NodeBundle {
+                    style: Style {
+                        width: Val::Percent(100.),
+                        height: Val::Percent(100.),
+                        padding: UiRect::all(Val::Px(20.)),
+                        ..default()
+                    },
                     ..default()
                 },
-            ));
-            buttons_panel(parent);
-        });
+            ))
+            .with_children(|parent| {
+                parent.spawn(TextBundle::from_section(
+                    *camera_name,
```

Comment:
Nittiest of nits: `format!("Player {}", index + 1)`

Reply:
I'd agree in regular code. I'll skip for here because it's probably more readable to have the full string in code when parsing the example and looking for cameras.

Reply:
I kinda like that the names are defined along with the positions in the for loop

Reply:
Sure